parteh/PRTGenericMod.F90

```diff
@@ -0,0 +1,1279 @@
+module PRTGenericMod
+
+  ! ------------------------------------------------------------------------------------
+  ! Plant Allocation and Reactive Transport (PART) +
+  ! Extensible Hypotheses (EH) = PARTEH
+  ! 
+  ! Non-Specific (Generic) Classes and Functions
+  ! This contains the base classes for both the variables and the global class
+  ! This also contains science relevent procedures that are agnostic of hypothesis
+  ! such as maintenance turnover and restranslocation.
+  !
+  ! THIS ROUTINE SHOULD NOT HAVE TO BE MODIFIED TO ACCOMODATE NEW HYPOTHESES
+  ! (in principle ...)
+  !
+  ! Ryan Knox, April 2018
+  ! ------------------------------------------------------------------------------------
+
+
+  use FatesConstantsMod, only : r8 => fates_r8
+  use FatesConstantsMod, only : i4 => fates_int
+  use FatesConstantsMod, only : nearzero
+  use FatesConstantsMod, only : calloc_abs_error
+  use FatesGlobals     , only : endrun => fates_endrun
+  use FatesGlobals     , only : fates_log 
+  use shr_log_mod      , only : errMsg => shr_log_errMsg
+ 
+  
+  implicit none
+
+  integer, parameter :: maxlen_varname   = 128
+  integer, parameter :: maxlen_varsymbol = 32
+  integer, parameter :: maxlen_varunits  = 32
+  integer, parameter :: len_baseunit     = 6
+
+  ! SEND THESE TO CONSTANTS
+
+  ! We use this parameter as the value for which we set un-initialized values
+  real(r8), parameter :: un_initialized = -9.9e32_r8
+
+  ! We use this parameter as the value for which we check un-initialized values
+  real(r8), parameter :: check_initialized = -8.8e32_r8
+
+
+  ! -------------------------------------------------------------------------------------
+  ! IMPORTANT!
+  ! All species in all organs should be expressed in terms of KILOGRAMS
+  ! All rates of change are expressed in terms of kilorams / day
+  ! This assumption cannot be broken!
+  ! -------------------------------------------------------------------------------------
+  
+  character(len=len_baseunit), parameter :: mass_unit = 'kg'
+  character(len=len_baseunit), parameter :: mass_rate_unit = 'kg/day'
+
+  ! -------------------------------------------------------------------------------------
+  ! Allocation Hypothesis Types
+  ! These should each have their own module
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: prt_carbon_allom_hyp   = 1
+  integer, parameter :: prt_cnp_flex_allom_hyp = 2   ! Still under development
+
+
+  ! -------------------------------------------------------------------------------------
+  ! Organ types
+  ! These are public indices used to map the organs
+  ! in each hypothesis to organs that acknowledged in the calling model
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: num_organ_types = 6
+  integer, parameter :: all_organs    = 0    ! index for all organs
```

Comment:
is this all_organs index ever used anywhere?

Reply:
no, but this index would be used for that function you suggested that handles the calculation of masses in organ sets, such as total biomass, or total live biomass.